win32ss/gdi/gdi32/objects/bidi.c

```diff
@@ -0,0 +1,687 @@
+/*
```

Comment:
Since this file comes from Wine, maybe you need to add an entry in [README.WINE](https://git.reactos.org/?p=reactos.git;a=blob;f=media/doc/README.WINE;h=6dc5502c87b69dbbf9ee1bd984ee4f91cc60a877;hb=HEAD#l266) to specify with which Wine version this file has been synced.
Also I would suggest, if you have made changes there, to surround these using some `#ifdef __REACTOS__ ... #endif` as we do for the synced files/modules, *unless* this file has too many changes with respect to the original and is basically a fork.

Reply:
For that file i only trimmed the common windows header inclusions, and commented out the 2 internal wine headers that are not being used in ros.